temboardagent/notification.py

```diff
@@ -1,64 +1,116 @@
-from temboardagent.queue import Queue, Message
-from temboardagent.errors import NotificationError
-import json
-import datetime
+# coding: utf-8
 
-"""
-Notifications are messages stored in a Queue aimed to keep a track of each
-possible action impacting PostgreSQL server configuration or behaviour.
+from datetime import datetime
+import logging
+import os
+import sqlite3
+from textwrap import dedent
 
-For now, they are pushed when one of the following actions is triggered:
-  * PostgreSQL reload, stop, start, restart.
-  * Setting changes.
-  * HBA file update.
+from temboardagent.errors import NotificationError
+from .tools import now
 
-Queue max size is set to 10MB, LIFO behaviour is expected.
-"""
+logger = logging.getLogger(__name__)
 
 
 class Notification(object):
+    """Notifications are log messages stored in a sqlite table aimed to keep a
+    track of each action impacting PostgreSQL server configuration or
+    behaviour.
+
+    For now, they are pushed when one of the following actions is triggered:
+        * PostgreSQL reload
+        * setting change
+        * user login
+        * user session expires
+        * PostgreSQL backend termination
+
+    Maximum number of record is 1000, LIFO behaviour is expected.
+    """
 
     def __init__(self, username, message, date=None):
+
         if date is None:
-            self.date = datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%S")
-        else:
-            self.date = date
+            # Set date to NOW UTC and format it as ISO8601 standard
+            date = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
         if type(username) == bytes:
             username = username.decode('utf-8')
         if type(message) == bytes:
             message = message.decode('utf-8')
+
+        self.time = now()
+        self.date = date
         self.username = username
         self.message = message
 
 
 class NotificationMgmt():
 
+    @classmethod
+    def bootstrap(self, config):
+        db_path = os.path.join(config.temboard.home, 'core.db')
+        with sqlite3.connect(db_path) as conn:
+            c = conn.cursor()
+            c.execute(
+                dedent("""
+                    CREATE TABLE IF NOT EXISTS action_logs (
+                        time REAL,
+                        date TEXT,
```

Comment:
Storing both `time` and `date` (a datetime ISO8601-formatted) seems redundant. Perhaps we could only store `time` and convert to a datetime when needed (in `get_last_n()` below)?

Reply:
Done.